### PR TITLE
[MU4] Fix some MSVC compiler warnings

### PR DESCRIPTION
**Pull request by Jojo-Schmitz**
and disable some for third party code

### CODE REVIEW

File: framework/audio/devtools/audioenginedevtools.cpp [hunk 1]
```diff
@@ -116,7 +116,7 @@ void AudioEngineDevTools::makeArpeggio()
                          chunk.beginTick = tick;
                          chunk.endTick = chunk.beginTick + duration;
 
-                         uint32_t note_duration = duration / notes.size();
+                         uint32_t note_duration = uint32_t(duration / notes.size());
```
Comment (Eism) [inline framework/audio/devtools/audioenginedevtools.cpp:119]: please, do not use old style type conversion, use static_cast

File: framework/midi/internal/soundfontsprovider.cpp [hunk 2]
```diff
@@ -107,8 +107,8 @@ std::vector<io::path> SoundFontsProvider::soundFontPaths(SoundFontFormats format
             continue;
         }
 
-        for (const io::path& path : files.val) {
-            soundFonts.push_back(path);
+        for (const io::path& pathf : files.val) {
```
Comment (Eism) [inline framework/midi/internal/soundfontsprovider.cpp:110]: better to name as filePath

Comment (Jojo-Schmitz) [reply framework/midi/internal/soundfontsprovider.cpp:110]: done

File: framework/audio/devtools/audioenginedevtools.cpp (hunk 1 above)
Comment (Jojo-Schmitz) [reply framework/audio/devtools/audioenginedevtools.cpp]: done

File: framework/audio/internal/audioplayer.cpp [hunk 3]
```diff
@@ -62,7 +62,7 @@ void AudioPlayer::setMidiStream(const std::shared_ptr<midi::MidiStream>& stream)
 
         m_tracks.clear();
         for (size_t num = 0; num < stream->initData.tracks.size(); ++num) {
-            m_tracks[num] = std::make_shared<Track>();
+            m_tracks[int(num)] = std::make_shared<Track>();
```
Comment (Eism) [inline framework/audio/internal/audioplayer.cpp:65]: and here too :)

Comment (Jojo-Schmitz) [reply framework/audio/internal/audioplayer.cpp:65]: done ;-)

**Review by Eism — APPROVED** [review]

**Pull request merged**